patterns/compose-structs.md

```diff
@@ -0,0 +1,101 @@
+# Compose structs together for better borrowing
```

Comment:
Maybe "Decompose struct for better borrowing" would be better.  The act, if one were refactoring, here seems to be one of decomposition rather than composition.  Saying "Compose ... together" is redundant.